plugins/grpc/cpp/v1.49.1/Dockerfile

```diff
@@ -0,0 +1,23 @@
+# syntax=docker/dockerfile:1.4
+FROM debian:bullseye-20221004 AS build
+
+ARG TARGETARCH
+
+RUN apt-get update \
+ && apt-get install -y curl git cmake build-essential autoconf clang libc++-dev libtool pkg-config g++ unzip zip
+RUN arch=${TARGETARCH}; \
+    if [ "${arch}" = "amd64" ]; then arch="x86_64"; fi; \
+    curl -fsSL -o /usr/local/bin/bazel https://github.com/bazelbuild/bazel/releases/download/4.2.2/bazel-4.2.2-linux-${arch} \
+ && chmod +x /usr/local/bin/bazel
+
+WORKDIR /build
+
+RUN git clone --depth 1 --branch v1.49.1 --recursive https://github.com/grpc/grpc
+WORKDIR /build/grpc
+RUN bazel build //src/compiler:grpc_plugin_support
+RUN bazel build //src/compiler:grpc_cpp_plugin
```

Comment:
Updated to build grpc plugins using Bazel instead of cmake (recommended in grpc development docs). This also allows us to benefit from caching since we'll only build the `:grpc_plugin_support` rule once and then have to build a single binary in each of the plugins (uncached).